src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java

```diff
@@ -407,26 +416,73 @@ public void onBestBlock_return_when_node_is_syncing() throws BtcReleaseClientExc
         ethereumListener.get().onBestBlock(null, null);
 
         // Assert
-        Mockito.verify(federatorSupport, Mockito.never()).getStateForFederator();
+        verify(federatorSupport, never()).getStateForFederator();
     }
 
     @Test
-    public void onBlock_return_when_node_is_syncing() {
+    public void onBestBlock_return_when_pegout_is_disabled() throws BtcReleaseClientException {
         // Arrange
         Federation federation = TestUtils.createFederation(params, 1);
 
         Ethereum ethereum = mock(Ethereum.class);
         AtomicReference<EthereumListener> ethereumListener = new AtomicReference<>();
 
-        Mockito.doAnswer((InvocationOnMock invocation) -> {
+        doAnswer((InvocationOnMock invocation) -> {
             ethereumListener.set((EthereumListener) invocation.getArguments()[0]);
             return null;
         }).when(ethereum).addListener(ArgumentMatchers.any(EthereumListener.class));
 
         FederatorSupport federatorSupport = mock(FederatorSupport.class);
 
         FedNodeSystemProperties fedNodeSystemProperties = mock(FedNodeSystemProperties.class);
-        Mockito.doReturn(Constants.regtest()).when(fedNodeSystemProperties).getNetworkConstants();
+        doReturn(Constants.regtest()).when(fedNodeSystemProperties).getNetworkConstants();
+        doReturn(false).when(fedNodeSystemProperties).isPegoutEnabled();
+
+        NodeBlockProcessor nodeBlockProcessor = mock(NodeBlockProcessor.class);
+        when(nodeBlockProcessor.hasBetterBlockToSync()).thenReturn(false);
+
+        BtcReleaseClient btcReleaseClient = new BtcReleaseClient(
+            ethereum,
+            federatorSupport,
+            fedNodeSystemProperties,
+            nodeBlockProcessor
+        );
+        btcReleaseClient.setup(
+            mock(ECDSASigner.class),
+            mock(ActivationConfig.class),
+            mock(SignerMessageBuilderFactory.class),
+            mock(ReleaseCreationInformationGetter.class),
+            mock(ReleaseRequirementsEnforcer.class),
+            mock(BtcReleaseClientStorageAccessor.class),
+            mock(BtcReleaseClientStorageSynchronizer.class)
+        );
+        btcReleaseClient.start(federation);
+
+        // Act
+        ethereumListener.get().onBestBlock(null, null);
+
+        // Assert
+        verify(federatorSupport, never()).getStateForFederator();
```

Comment:
Let's add one more assertion

```suggestion
        verify(nodeBlockProcessor, times(1)).hasBetterBlockToSync();
        verify(federatorSupport, never()).getStateForFederator();
```